moto/autoscaling/models.py

```diff
@@ -609,7 +679,21 @@ def make_int(value):
             health_check_period = 300
         else:
             health_check_period = make_int(health_check_period)
-        if launch_config_name is None and instance_id is not None:
+
+        # TODO: Add MixedInstancesPolicy once implemented.
+        # Verify only a single launch config-like parameter is provided.
+        count = 0
```

Comment:
This can be a one-liner: `nr_of_parameters = sum([1 for prm in [launch_config_name, launch_template, instance_id if prm])`